### ISSUE TITLE
Probable bug: ensure this behaves properly with 'instanceof(...)' in this scenario.

### ISSUE THREAD

**Issue by stingus** [kalessil/phpinspectionsea#671]
I have no idea what's wrong in a construct like this or how should I make it right, but it throws `Probable bug: ensure this behaves properly with 'instanceof(...)' in this scenario.`:
```php
if (true === $x && $y instanceof MyClass) {
    ...
}
```
EA is marking `$y instanceof MyClass` as weak warning. The only case when this warning is not set is when `instanceof` is the only condition. This started after upgrading to PHPStorm 2017.3 and in 2017.2.4 it worked fine.

**Comment by kalessil** [comment]
@stingus thank you for reporting. I can not reproduce, unfortunately, can you share bigger code fragment (GitHub link, simplified function/method) please?

**Comment by Bob4ever** [comment]
Here is an example:
```php
<?php

class A
{
}

class SuperA extends A
{
    public function asdf()
    {
        return true;
    }
}


function test(A $a)
{
    if ($a instanceof SuperA && $a->asdf()) {
        return 1;
    }
    return null;
}

```

**Comment by kalessil** [comment]
Confirmed, 2017.3 specific. Investigating.

**Comment by kalessil** [comment]
Fixed 2.3.13.2 is coming today
